### PR TITLE
Update Core/Security/Support/SMA Service Dockerfiles for Go 1.13.

### LINKED ISSUE #2341: Update Core/Support/SMA Service Dockerfiles for Go 1.13
A Geneva release goal is to update our dependency on Go to utilize v1.13 of the compiler. For this issue, please modify all core/support Dockerfiles to use Go 1.13.

Example of current reference in core-command service:
https://github.com/edgexfoundry/edgex-go/blob/f0ecddb33a38178a7f9f83b66f655389afed08a7/cmd/core-command/Dockerfile#L9

Change the reference from `1.12` to `1.13` for all core, support and SMA-related images.

Please also change the `go 1.12` reference in the `go.mod` to `go 1.13`

### PR DESCRIPTION
**Pull request by akramtexas**
Fix #2341 
- After making the updates to the Dockerfiles, as well as to go.mod, I built the Docker images and made sure the Docker containers are launched properly. Also did a sanity check by hitting the `ping` endpoint each of all services for the correct response.

### CODE REVIEW

**Comment by akramtexas** [comment]
Rebased

**Comment by akramtexas** [comment]
> Missed references in https://github.com/edgexfoundry/edgex-go/blob/master/README.md, https://github.com/edgexfoundry/edgex-go/blob/master/snap/snapcraft.yaml

- Addressed the above comment, and also rebased. I found the references for snapcraft.yaml in: https://golang.org/dl/
```
go1.13.7.linux-amd64.tar.gz Archive Linux x86-64  115MB b3dd4bd781a0271b33168e627f7f43886b4c5d1c794a4015abf34e99c6526ca3
go1.13.7.linux-arm64.tar.gz Archive Linux ARMv8 93MB  8717de6c662ada01b7bf318f5025c046b57f8c10cd39a88268bdc171cc7e4eab
go1.13.7.linux-armv6l.tar.gz  Archive Linux ARMv6 94MB  ff8b870222d82c38a0108810706811dcbd1fcdbddc877789184a0f903cbdf11a
go1.13.7.linux-386.tar.gz Archive Linux x86 97MB  93e82683f32d9fe7fda9b686415aeee599a92c4e450b69519bb53e1d62144a85
```

**Comment by akramtexas** [comment]
> You updated snap-related but not README.md.

- That was an oversight on my part. Addressed, thanks.

**Comment by akramtexas** [comment]
Recheck - Had got the following error:
```
10:40:50 go: downloading github.com/magiconair/vendorfmt v0.0.0-20170905193701-0fde667441eb
10:40:50 go: extracting github.com/magiconair/vendorfmt v0.0.0-20170905193701-0fde667441eb
10:40:51 go get gopkg.in/matm/v1/gocov-html: unrecognized import path "gopkg.in" (parse https://gopkg.in/?go-get=1: no go-import meta tags ())
10:40:51 GNUmakefile:235: recipe for target 'tools' failed
10:40:51 make: *** [tools] Error 1
10:40:52 Downloading 'openresty-1.15.8.1.tar.gz'   0%
```

**Comment by akramtexas** [comment]
Recheck

File: README.md [hunk 1]
```diff
@@ -50,7 +50,7 @@ on the snap, including how to install it, please refer to [EdgeX snap](https://g
 
 #### Go
 
-- The current targeted version of the Go language runtime for release artifacts is v1.12.x
+- The current targeted version of the Go language runtime for release artifacts is v1.13.x
 - The minimum supported version of the Go language runtime is v1.11.x (currently v1.11.13)
```
Comment (michaelestrin) [inline README.md]: Minimum is also 1.13.

Comment (akramtexas) [reply README.md]: Updated `README.md` for minimum version 1.13 ✅

**Comment by akramtexas** [comment]
Rebased

**Comment by michaelestrin** [comment]
@akramtexas 

snap builds failing.

image

**Comment by akramtexas** [comment]
> @akramtexas
> 
> snap builds failing.
> 
> image

I'm investigating 👍

**Comment by akramtexas** [comment]
Rebased

**Comment by akramtexas** [comment]
- Continuing to investigate why the snap builds are failing... Looked into [the Snapcraft.yaml reference](https://snapcraft.io/docs/snapcraft-yaml-reference), too, but no headway gained thereby. 
- I reached out to @tonyespy yesterday evening for pointers and he graciously tried a local build, and it worked just fine. He is thinking that there may be something else happening on the build side of things. Tony offered to and plans on taking a closer look first thing today 👍

**Comment by akramtexas** [comment]
Rebased

**Comment by akramtexas** [comment]
@tonyespy has been working this `snap`-related issue in concert with a few others as well, and managed to reproduce this issue. Thanks 👍 The issue is unique to the way the `snap` is built on Jenkins (i.e. they build the `snap` in the `Docker` container, because the LF doesn't run `Ubuntu` for all of the builds). I will update here as soon as they have a fix.

**Comment by akramtexas** [comment]
Rebased

**Comment by akramtexas** [comment]
@tonyespy and team have made some progress, but don't have a root-cause yet. Adding further details from Tony: _"It seems related to building the old version (1.3.x) version of consul using go 1.13. When deployed with docker, we use a standard consul docker image, whereas with the snap, we build consul from source. (Ian spent a few hours on this earlier today)."

**Comment by akramtexas** [comment]
Rebased

**Comment by tonyespy** [comment]
Just to be clear, I'm fine with you backing out the changes to snapcraft.yaml in your PR. As you mentioned above, I'm already working on this, and can make the changes to the snap once I've worked out the root cause.

**Comment by akramtexas** [comment]
> Just to be clear, I'm fine with you backing out the changes to snapcraft.yaml in your PR. As you mentioned above, I'm already working on this, and can make the changes to the snap once I've worked out the root cause.

Yes, thanks @tonyespy 👍 - I will shortly back-out the changes to snapcraft.yaml in this PR (and create a _separate_ PR which will include the changes to snapcraft.yaml, and whose `snap` builds should pass once the issue has been fixed). For easy tracking, mentioning the issue that I have created for addressing this build failure, as follows: [Issue-2378](https://github.com/edgexfoundry/edgex-go/issues/2378)

**Comment by michaelestrin** [comment]
Please address the build failure so this can be reviewed/merged.

image

@akramtexas

**Comment by akramtexas** [comment]
> Please address the build failure so this can be reviewed/merged.
> 
> image
> 
> @akramtexas

I'm on it 👍

**Comment by akramtexas** [comment]
> Please address the build failure so this can be reviewed/merged.
> 
> image
> 
> @akramtexas

@michaelestrin - Done ✅

**Pull request merged**